enhancements/network/admin-network-policy.md

```diff
@@ -0,0 +1,322 @@
+
+```yaml
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: AdminNetworkPolicy
+metadata:
+  name: priority-50-example
+spec:
+  priority: 50
+  subject:
+    namespaces:
+      matchLabels:
+          kubernetes.io/metadata.name: gryffindor
+  ingress:
+  - name: "deny-all-ingress-from-slytherin"
+    action: "Deny"
+    from:
+    - pods:
+        namespaces:
+          namespaceSelector:
+            matchLabels:
+              conformance-house: slytherin
+        podSelector:
+          matchLabels:
+            conformance-house: slytherin
+  egress:
+  - name: "pass-all-egress-to-slytherin"
+    action: "Pass"
+    to:
+    - pods:
+        namespaces:
+          namespaceSelector:
+            matchLabels:
+              conformance-house: slytherin
+        podSelector:
+          matchLabels:
+            conformance-house: slytherin
+```
+
+The OVN-K controller will convert these into NBDB logical entities (showing only egress, but same will be done for ingress):
+
+```text
+_uuid               : b29f0a44-30b2-4857-985f-a2e899ea72cd
+acls                : [c02b4254-b1d0-48c1-a091-57ae6887bdc3, d4a487dd-3ac0-44d7-b07d-8d6eb0729567]
+external_ids        : {name=priority-50-example_25000}
+name                : a16480371363466183296
+ports               : []
+```
+```text
+_uuid               : 88a1905f-9733-479d-a23e-4e19aab9f95a
+addresses           : []
+external_ids        : {direction=ANPEgress, ip-family=v4, "k8s.ovn.org/id"="network-policy-controller:AdminNetworkPolicy:priority-50-example:ANPEgress:25000:v4", "k8s.ovn.org/name"=priority-50-example, "k8s.ovn.org/owner-controller"=network-policy-controller, "k8s.ovn.org/owner-type"=AdminNetworkPolicy, priority="25000"}
+name                : a11410890234882820008
+```
+```text
+_uuid               : d4a487dd-3ac0-44d7-b07d-8d6eb0729567
+action              : pass
+direction           : from-lport
+external_ids        : {direction=ANPEgress, "k8s.ovn.org/id"="network-policy-controller:AdminNetworkPolicy:priority-50-example:ANPEgress:25000", "k8s.ovn.org/name"=priority-50-example, "k8s.ovn.org/owner-controller"=network-policy-controller, "k8s.ovn.org/owner-type"=AdminNetworkPolicy, priority="25000"}
+label               : 0
+log                 : false
+match               : "((ip4.dst == $a11410890234882820008)) && (inport == @a16480371363466183296)"
+meter               : acl-logging
+name                : priority-50-example_ANPEgress_25000
+options             : {apply-after-lb="true"}
+priority            : 25000
+severity            : debug
+tier                : 1
+```
+
+#### Variation [optional]
+
+N/A
+
+### API Extensions
+
+None
+
+### Implementation Details/Notes/Constraints [optional]
+
+Covered in the above sections
+
+### Risks and Mitigations
+
+N/A
+
+### Drawbacks
+
+N/A
+
+## Design Details
+
+### Open Questions [optional]
+
+Aligning existing network policy V1 code with this new controller -
```

Comment:
Was this not decided already?  Is seems like level driven is in-fact the "right" way to do things as it pertains to Kubernetes controllers.

Reply:
yes I will remove this block actually. It is not relevant anymore.